tests/ui/methods.rs

```diff
@@ -1,7 +1,13 @@
 // aux-build:option_helpers.rs
 // compile-flags: --edition 2018
 
-#![warn(clippy::all, clippy::pedantic, clippy::option_unwrap_used)]
+#![warn(
+    clippy::all,
+    clippy::pedantic,
+    clippy::option_unwrap_used,
+    clippy::option_expect_used,
+    clippy::result_expect_used
```

Comment:
Can you put the `expect` tests in their own file? See #2038

Reply:
i can, i'm a bit unsure what the methodology is, from the tests I looked at it usually was "file per lint test" and tried to stick that, i.e. the `expect` tests are in methods as they're in the lint pass for methods, the `unreachable` is in the panic_unimplemented as it's in that lint pass. I couldn't put `panic` in panic_unimplemented since it clashed with the other panic tests there but looking at methods.rs I could add it as a warning.

So should I put all the new lints into a new file or make one file each or combine them all into their lint passes?

Reply:
"one file per lint" is a goal, that we want to achieve, but we're not really there yet, as you can see in the `methods.rs` test file. (Also "one file per lint" can also mean grouping some of the lints, when it makes sense, e.g. the 4 panic macro `Lint`s) 

For clarification: `Lint != LintPass`. One file per `LintPass` would be too big, since for example the `methods` module uses a single `LintPass` to check X (X=big number, that I don't want to count) `Lint`s.

> So should I put all the new lints into a new file or make one file each or combine them all into their lint passes?

What I would suggest is to put the `expect` lint tests in one file and the panic macro tests in their own file. 

So for the currently implemented lints, you should create 2 test files:

1. `tests/ui/panicking_macros.rs`, which tests:
    - `panic`
    - `unimplemented` (Move this tests out of `panic_unimplemented.rs` into the new file; also maybe rename the `panic_unimplemented.rs` file)
    - `unreachable`
    - `todo`
2. `tests/ui/expect`, which tests:
    - `option_expect_used`
    - `result_expect_used`

---

Bonus points, if you also move the `unwrap` lints to their own file.